network-testing/src/main/java/com/stripe/android/networktesting/TestBodyFromFile.kt

```diff
@@ -33,7 +36,32 @@ fun MockResponse.testBodyFromFile(filename: String): MockResponse {
     val inputStream = MockResponse::class.java.classLoader!!.getResourceAsStream(filename)
     val buffer = Buffer()
     buffer.readFrom(inputStream)
+    assertIsValidJsonString(buffer.clone().readUtf8(), filename)
     setBody(buffer)
 
     return this
 }
+
+/**
+ * Validates JSON syntax and fails the test if the provided JSON string is not valid.
+ *
+ * This method attempts to parse the JSON string using JSONObject to detect syntax errors
+ * such as missing brackets, invalid commas, or malformed structure. Note that this validation
+ * only checks syntactic correctness, not semantic accuracy (e.g., it won't detect field name
+ * typos or missing required fields).
+ *
+ * @param json The JSON string to validate
+ * @param filename The source filename, used in error reporting for easier debugging
+ * @throws AssertionError If JSON parsing fails, with information about the source file
+ *
+ * Note: For detailed error diagnostics, examine the JSON file in Android Studio, which provides
+ * built-in JSON formatting and specific error detection capabilities.
+ */
+private fun assertIsValidJsonString(json: String, filename: String) {
+    try {
+        JSONObject(json)
+    } catch (_: JSONException) {
+        // MockWebServer catches the exception so we need an error to fail the test
+        throw AssertionError("Parsing JSON failed for file: $filename")
```

Comment:
I think we should actually just crash the process here (after throwing the exception)! 

`android.os.Process.killProcess(android.os.Process.myPid())`